src/modes/thread_view/webextension/ae_protocol.cc

```diff
@@ -139,10 +139,16 @@ namespace Astroid {
     gsize read = 0;
     bool  s    = false;
 
+    AeProtocol::MessageTypes mt;
+
     /* read message size */
     gsize msg_sz = 0;
     s = istream->read_all ((char *) &msg_sz, sizeof (msg_sz), read, reader_cancel);
 
+    if (s && read == 0) {
+      return mt;
```

Comment:
All in all, this looks very good. I'm not sure about this line, if the read function fails then I think that the caller should handle the exception more gracefully and just exit rather than receiving an empty MessageType. It is not clear to the calling thread whether the read call succeded or not.

Reply:
Yes, there seems to be more going on here. I regularly get log messages about broken pipes, so it seems that the read end is closed before all data has been pushed into the pipe on the write end and messages are lost. I plan to have a look at that.

Technically this case here isn't an exception; it covers the case where the stream has been closed. This is a perfectly valid case IMHO and shouldn't return an error.

Reply:
I agree, it is not an error. I just think that it should be handled by the reader thread which can then shut it self down, it could also be passed with a CloseStream messagetype.

Reply:
Ok, maybe postpone this until I've had a chance to look into why messages are dropped.